src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -97,12 +95,15 @@ def __init__(
         >>> pd.Index([1, 2, 3], dtype="uint8")
         Index([1, 2, 3], dtype='int64')
         """
-        self._parent = data if isinstance(data, BasePandasDataset) else None
-        data = data._query_compiler if isinstance(data, BasePandasDataset) else data
-        if isinstance(data, SnowflakeQueryCompiler):
-            qc = data
+        if query_compiler:
+            self._query_compiler = query_compiler
+        elif isinstance(data, BasePandasDataset):
+            self._parent = data
+            self._query_compiler = data._query_compiler.drop(
```

Comment:
I know this behavior existed before this PR, but it doesn't match pandas:

```python
df = pd.DataFrame([1, 2])
# in pandas, get ValueError: Index data must be 1-dimensional
# in Snowpark pandas, get Index([0, 1], dtype='int64')
print(pd.Index([1, 2]))
```

Is this behavior intentional? Do we document it?

Reply:
I cannot follow the above examples they should both work.

Reply:
Sorry, I edited my example. I meant constructing an index from the dataframe.

Reply:
oh, that should be a bug. Good catch! Let me see if I can fix it here.